apps/api/src/domain/models/tag/commands/tag-resource-or-note/resource-added-to-tag.event-handler.ts

```diff
@@ -0,0 +1,53 @@
+import { CategorizableCompositeIdentifier, CategorizableType } from '@coscrad/api-interfaces';
+import { Inject } from '@nestjs/common';
+import { CoscradEventConsumer, ICoscradEventHandler } from '../../../../../domain/common';
+import { AggregateId } from '../../../../../domain/types/AggregateId';
+import { Maybe } from '../../../../../lib/types/maybe';
+import { isNotFound } from '../../../../../lib/types/not-found';
+import { QUERY_REPOSITORY_PROVIDER_TOKEN } from '../../../shared/common-commands/publish-resource/resource-published.event-handler';
+import {
+    ITagQueryRepository,
+    TAG_QUERY_REPOSITORY_PROVIDER_TOKEN,
+} from '../../repositories/tag-query-repository.interface';
+import { ResourceOrNoteTagged } from './resource-or-note-tagged.event';
+
+interface QueryRepositoryForTaggable<T = unknown> {
+    fetchById(id: AggregateId): Promise<Maybe<T>>;
+}
+
+interface IQueryRepositoryProvider {
+    // TODO note test case
+    forResource<T = unknown>(categorizableType: CategorizableType): QueryRepositoryForTaggable<T>;
+}
+
+@CoscradEventConsumer('RESOURCE_OR_NOTE_TAGGED')
+export class ResourceAddedToTagEventHandler implements ICoscradEventHandler {
+    constructor(
+        @Inject(TAG_QUERY_REPOSITORY_PROVIDER_TOKEN)
+        private readonly repository: ITagQueryRepository,
+        @Inject(QUERY_REPOSITORY_PROVIDER_TOKEN)
+        private readonly categorizableRepositoryProvider: IQueryRepositoryProvider
+    ) {}
+
+    async handle({
+        payload: {
+            aggregateCompositeIdentifier: { id: tagId },
+            taggedMemberCompositeIdentifier: { type: categorizableType, id: categorizableId },
+        },
+    }: ResourceOrNoteTagged): Promise<void> {
+        // TODO wrap in a transaction
```

Comment:
Let's try this.

Reply:
The only risk of not making it a transaction is that there's a tiny chance that the resource could get removed and still be written to the tag document. Where transactions are more important is when multiple writes are called in a single handler. If the first one fails and the logic isn't idempotent, you can end up with invalid states that are worse than a missing reference from tag to resource.